src/validation/pou.rs

```diff
@@ -29,6 +29,22 @@ pub fn visit_implementation<T: AnnotationMap>(
     }
     if implementation.linkage != LinkageType::External {
         validate_action_container(validator, implementation);
+        //Validate the label uniquiness
+        if let Some(mut labels) = context.index.get_labels(&implementation.name).map(SymbolMap::values) {
+            if let Some(first) = labels.next() {
+                if let Some(second) = labels.next() {
+                    //Collect remaining
+                    let mut locations: Vec<_> = labels.map(|it| it.location.clone()).collect();
+                    locations.push(first.location.clone());
+                    locations.push(second.location.clone());
+                    validator.push_diagnostic(Diagnostic::duplicate_label(
+                        &first.name,
+                        first.location.clone(),
+                        locations,
+                    ));
+                }
+            }
+        }
```

Comment:
I might be missing something but will this not always evaluate to true (regardless of the labels name) if the implementation has a number of labels >1?